README.md

```diff
@@ -0,0 +1,34 @@
+# Scala Bindings for Materials Data Facility API
+This library contains a service and associated case classes for interacting
+with MDF's Connect service. This service is used to submit datasets
+for ingesting into the [facility](https://materialsdatafacility.org).
+
+# Prerequisites
+In order to use this library you will require a client id and secret. This
+can be obtained from [Globus](https://www.globus.org). You will then need to
+contact the MDF team to grant your account permission to publish via the API.
```

Comment:
Why does the user need a confidential client? MDF Connect requires a user to be authenticated, but they can use their own credentials (anything recognized by GlobusAuth, including many research institutions, Google, ORCiD, and Globus ID). Additionally, I think the language here could be clearer about the "account permission" being acceptance into the [MDF Connect Convert](https://www.globus.org/app/groups/cc192dca-3751-11e8-90c1-0a7c735d220a/about) Globus Group, which will likely not require MDF admin approval after the official Connect public launch.

Reply:
My focus now is to support 4CeeD publication which uses a confidential client. We can extend the library to support other permission models in future releases when we have use cases to test out

Reply:
I'm just concerned that this package is more cumbersome to use than the Python package for the standard user, and could confuse someone into thinking that it's hard to get started with MDF. Perhaps if we tweak the readme to say that non-client permission flows are planned, but not yet implemented? Then we could still test the current implementation using the 4CeeD case, but not suggest that other must adhere to that flow.

Reply:
@jgaff - I updated the README as suggested. Thanks for the input.

Reply:
Great, that looks good to me. I'd call that ready to merge, unless you have further refinements you want to make.